src/Build/BuildCheck/Checks/UntrustedLocationCheck.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.Experimental.BuildCheck.Checks;
+internal sealed class UntrustedLocationCheck : Check
```

Comment:
Should we save this generic name is we check Downloads folder only?

Reply:
The check description and error message are more concrete.
So probably it shouldn't matter that name is bit more generic? (and we might possibly expand the checking in the future - e.g. for locations with too permissive access; root drive location; etc.)